src/NHibernate/Async/Criterion/QueryOver.cs

```diff
@@ -0,0 +1,133 @@
+	{
+
+		private Task<IList<TRoot>> ListAsync(CancellationToken cancellationToken)
+		{
+			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<IList<TRoot>>(cancellationToken);
+			}
+			return criteria.ListAsync<TRoot>(cancellationToken);
+		}
+
+		private Task<IList<U>> ListAsync<U>(CancellationToken cancellationToken)
+		{
+			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<IList<U>>(cancellationToken);
+			}
+			return criteria.ListAsync<U>(cancellationToken);
+		}
+
+		private Task<TRoot> SingleOrDefaultAsync(CancellationToken cancellationToken)
+		{
+			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<TRoot>(cancellationToken);
+			}
+			return criteria.UniqueResultAsync<TRoot>(cancellationToken);
+		}
+
+		private Task<U> SingleOrDefaultAsync<U>(CancellationToken cancellationToken)
+		{
+			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<U>(cancellationToken);
+			}
+			return criteria.UniqueResultAsync<U>(cancellationToken);
+		}
+
+
+		Task<IList<TRoot>> IQueryOver<TRoot>.ListAsync(CancellationToken cancellationToken)
+		{ 			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<
+
+		IList<TRoot>>(cancellationToken);
+			}
+return ListAsync(cancellationToken); }
+
+		Task<IList<U>> IQueryOver<TRoot>.ListAsync<U>(CancellationToken cancellationToken)
+		{ 			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<
+		IList<U>>(cancellationToken);
+			}
+return ListAsync<U>(cancellationToken); }
+
+		Task<int> IQueryOver<TRoot>.RowCountAsync(CancellationToken cancellationToken)
+		{ 			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<
+		int>(cancellationToken);
+			}
+			try
+			{
+return ToRowCountQuery().SingleOrDefaultAsync<int>(cancellationToken); 			}
+			catch (Exception ex)
+			{
+				return Task.FromException<int>(ex);
+			}
+}
+
+		Task<long> IQueryOver<TRoot>.RowCountInt64Async(CancellationToken cancellationToken)
+		{ 			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<
+		long>(cancellationToken);
+			}
+			try
+			{
+return ToRowCountInt64Query().SingleOrDefaultAsync<long>(cancellationToken); 			}
+			catch (Exception ex)
+			{
+				return Task.FromException<long>(ex);
+			}
+}
+
+		Task<TRoot> IQueryOver<TRoot>.SingleOrDefaultAsync(CancellationToken cancellationToken)
+		{ 			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<
+		TRoot>(cancellationToken);
+			}
+return SingleOrDefaultAsync(cancellationToken); }
+
+		Task<U> IQueryOver<TRoot>.SingleOrDefaultAsync<U>(CancellationToken cancellationToken)
+		{ 			if (cancellationToken.IsCancellationRequested)
+			{
+				return Task.FromCanceled<
+		U>(cancellationToken);
+			}
+return SingleOrDefaultAsync<U>(cancellationToken); }
```

Comment:
Weird formatting here. It looks like a glitch either with explicit interface implementation, or the special way it is formatted in synchronous code:
```c#
		TRoot IQueryOver<TRoot>.SingleOrDefault()
		{ return SingleOrDefault(); }
```